File: app/src/main/java/io/github/marcocipriani01/telescopetouch/search/SearchResult.java
/*
 * Copyright 2021 Marco Cipriani (@marcocipriani01)
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package io.github.marcocipriani01.telescopetouch.search;

import androidx.annotation.NonNull;

import io.github.marcocipriani01.telescopetouch.astronomy.GeocentricCoordinates;

/**
 * A single search result.
 *
 * @author John Taylor
 */
public class SearchResult {
    /**
     * The coordinates of the object.
     */
    public final GeocentricCoordinates coords;
    /**
     * The user-presentable name of the object, properly capitalized.
     */
    public final String capitalizedName;

    /**
     * @param capitalizedName The user-presentable name of the object, properly capitalized.
     * @param coords          The coordinates of the object.
     */
    public SearchResult(String capitalizedName, GeocentricCoordinates coords) {
        this.capitalizedName = capitalizedName;
        this.coords = coords;
    }

    @NonNull
    @Override
    public String toString() {
        return capitalizedName;
    }
}

File: app/src/main/java/com/google/android/stardroid/search/SearchResult.kt
// Copyright 2008 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
package com.google.android.stardroid.search

import com.google.android.stardroid.math.Vector3

/**
 * A single search result.
 *
 * @author John Taylor
 */
class SearchResult
/**
 * @param capitalizedName The user-presentable name of the object, properly capitalized.
 * @param coords The geocentric coordinates of the object.
 */(
    /** The user-presentable name of the object, properly capitalized. */
    var capitalizedName: String,
    /** The coordinates of the object. */
    var coords: Vector3?
) {
    override fun toString(): String {
        return capitalizedName
    }
}